resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/Comparators.java

```diff
@@ -325,6 +328,11 @@ public int compare(ResourceAddress addr1, ResourceAddress addr2) {
                 return compareLocation;
             }
 
+            int compareAlternateAndTransportLocation = ALTERNATE_AND_TRANSPORT_LOCATION_COMPARATOR.compare(addr1, addr2);
+            if(compareAlternateAndTransportLocation != 0) {
+               return compareAlternateAndTransportLocation;
```

Comment:
Here we compare the alternates of the transport for ssl (which are TcpResourceAddresses, with or without alternates), and the alternates of TcpResourceAddresses. This is somewhat inline with what equals from ResourceAddress does (if one has alternate and the other doesn't), and also check alternates agains each other using the AlternateResourceAddressComparator).